### ISSUE TITLE
`auctex-latexmk` cannot be checked out by github fetcher because of non-existent directory `.config/emacs/.cache/quelpa/build/`

### ISSUE THREAD

**Issue by bryce-carson** [syl20bnr/spacemacs#16653]
#### Description :octocat:
Recently I opened #16647, which described an issue caused by package updates
affecting magit. Another user experiencing the issue, who did not use the same
work-around as I did, suggested updating packages again. I thought to do that,
and begin keeping my packages up-to-date (I usually let them get quite stale).

Without removing the work-around I described---which shouldn't affect anything
because it is a valid user configuration option regardless---I updated my Emacs
packages. The message seen in [reproduction-guide] was then observed.

#### Reproduction guide :beetle:
- Start Emacs
- SPC f e U (`configuration-layer/update-packages`)

*Observed behaviour:* :eyes: :broken_heart:
The following message is displayed in *Messages*:
```
Fetcher: github
Source: smile13241324/auctex-latexmk

condition-case: Failed to checkout ‘auctex-latexmk’: ‘Can’t run process in non-existent directory: /home/bryce/.config/emacs/.cache/quelpa/build/’
```

*Expected behaviour:* :heart: :smile: auctex-latexmk should not fail to checkout
due to a non-existent directory; the directory, if required, should be created
(especially inside the Spacemacs cache directory).

#### System Info :computer:
- OS: gnu/linux
- Emacs: 29.4
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. 60dcd8cf9)
- Graphic display: t
- Running in daemon: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(csv html sql windows-scripts
     (python :variables python-format-on-save t)
     meson yaml
     (auto-completion :variables auto-completion-idle-delay 0.7 auto-completion-minimum-prefix-length 0 auto-completion-enable-snippets-in-popup t auto-completion-enable-help-tooltip t)
     emacs-lisp common-lisp
     (git :packages magit magit-section evil-collection evil-surround git-timemachine transient)
     themes-megapack helm
     (ess :variables ess-assign-key "\255" ess-r-backend nil)
     markdown info
     (java :variables java-backend 'lsp)
     (latex :variables latex-view-pdf-in-split-window t latex-refresh-preview t latex-build-engine 'luatex)
     org javascript
     (shell :variables shell-default-height 30 shell-default-position 'bottom shell-default-shell 'vterm)
     syntax-checking)
```
- System configuration features: ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GSETTINGS HARFBUZZ JPEG JSON LIBOTF LIBSELINUX LIBSYSTEMD LIBXML2 M17N_FLT MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND SQLITE3 THREADS TIFF TOOLKIT_SCROLL_BARS TREE_SITTER WEBP X11 XDBE XIM XINPUT2 XPM XWIDGETS GTK3 ZLIB

#### Backtrace :paw_prints:
*NOT APPLICABLE*

**Comment by bryce-carson** [comment]
Manually creating the directory allows package upgrade to continue, without any apparent error.

**Comment by smile13241324** [comment]
You are right, now Spacemacs will pre-create the given quelpa-build-dir during quelpas configuration phase.
